source/developers-guide/legacy-plugin-system/index.md

```diff
@@ -0,0 +1,820 @@
+
+* `custom/plugins/SwagResource/Resources/frontend/css/**.css`
+* `custom/plugins/SwagResource/Resources/frontend/js/**.js`
+* `custom/plugins/SwagResource/Resources/frontend/less/all.less`
+
+The `css` and `js` directories may contain arbitrary sub directories. `@import`s in `all.less` will be resolved.
+
+```php
+public function install()
+{
+    $this->subscribeEvent('Theme_Compiler_Collect_Plugin_Less', 'addLessFiles');
+    $this->subscribeEvent('Theme_Compiler_Collect_Plugin_Javascript', 'addJsFiles');
+}
+
+/**
+ * @return ArrayCollection
+ */
+public function addLessFiles()
+{
+    return new Doctrine\Common\Collections\ArrayCollection([
+        new LessDefinition(
+            [],
+            [__DIR__ . '/Views/frontend/_public/src/less/all.less'],
+            __DIR__
+        ),
+    ]);
+}
+
+/**
+ * @return ArrayCollection
+ */
+public function addJsFiles()
+{
+    return new ArrayCollection([
+        __DIR__ . '/Views/frontend/_public/src/js/jquery.swag_live_shopping.js',
+    ]);
+}
+```
+
+## Add console commands
+For register a console command create a command file which extends the class "ShopwareCommand".
+The register the event `Shopware_Console_Add_Command` and return a new ArrayCollection with your created commands.
+
+```php
+$this->subscribeEvent(
+    'Shopware_Console_Add_Command',
+    'onAddConsoleCommand'
+);
+
+/**
+ * Adds the console commands
+ *
+ * @return ArrayCollection
+ */
+public function onAddConsoleCommand()
+{
+    return new ArrayCollection(
+        [
+          new Command1(),
+          new Command2(),
+        ]
+    );
+}
+```
+
+```php
+<?php
+
+class ExportCommand extends ShopwareCommand
+{
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        
+    }
+    
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        
+    }
+}
+```
+
+## Add backend emotion components
+For creating custom shopping world elements Shopware provides some helper functions which can be used in the Bootstrap.php of a Shopware plugin. So all you have to do is to create a simple plugin where you can register one ore more elements via the createEmotionComponent() method. As an example for this tutorial we will create a Vimeo element for adding videos to the shopping world.
+ 
+```php
+$vimeoElement = $this->createEmotionComponent([
+    'name' => 'Vimeo Video',
+    'xtype' => 'emotion-components-vimeo',
+    'template' => 'emotion_vimeo',
+    'cls' => 'emotion-vimeo-element',
+    'description' => 'A simple vimeo video element for the shopping worlds.'
+]);
+```
+In the install() method of our plugin we register a new element and save it in the variable $vimeoElement for later use. The createEmotionComponent() method expects a configuration array with the following properties:
```

Comment:
here is something missing after following properties

Reply:
not missing